### PR TITLE
Improve init git hook script

### PR DESCRIPTION
**Pull request by williamboman**
- Prevent word splitting
- Only create new global templatedir if it's missing
- Append our hook script if hook file already exists.

### CODE REVIEW

**Comment by IonicaBizau** [comment]
Thanks! Add yourself in `contributors` array.

**Comment by williamboman** [comment]
Done :sunglasses:

File: scripts/init-git-post-commit [hunk 1]
```diff
@@ -1,21 +1,41 @@
...
+		&& echo "Successfully set up git-stats hook at ${post_commit_path}." \
+		&& exit 0
+else
+	printf "\n%s\n" "$hook" >> "$post_commit_path" \
+		&& echo "Successfully set up git-stats hook at ${post_commit_path}." \
+		&& exit 0
+fi
 
-chmod +x $post_commit_path
+echo "Couldn't set up git-stats hook."
```
Comment (IonicaBizau) [inline scripts/init-git-post-commit]: I guess this will never been fired, right? Since you used `if (exp) { ... exit 0; } else { ... exit 0; }`.

Comment (williamboman) [inline scripts/init-git-post-commit]: Basically no, unless `printf` fails for whatever reason (i.e. `printf` binary not in `$PATH`).

File: scripts/init-git-post-commit [hunk 2]
```diff
@@ -1,21 +1,41 @@
...
 EOF
+);
+
+if [ ! -f "$post_commit_path" ]; then
+	printf "#!/bin/sh\n%s" "$hook" > "$post_commit_path" \
+		&& chmod +x "$post_commit_path" \
+		&& echo "Successfully set up git-stats hook at ${post_commit_path}." \
+		&& exit 0
+else
+	printf "\n%s\n" "$hook" >> "$post_commit_path" \
```
Comment (IonicaBizau) [inline scripts/init-git-post-commit]: ...I don't like this. I'd say to check if there is already something related to `git-stats` there:

```sh
$ grep 'git-stats' ~/.git-templates/hooks/post-commit
```
image

File: scripts/init-git-post-commit (hunk 1 above)
Comment (IonicaBizau) [inline scripts/init-git-post-commit]: Ah, I see. :+1:

File: scripts/init-git-post-commit (hunk 2 above)
Comment (williamboman) [inline scripts/init-git-post-commit]: Yeah my initial plan was to replace any previous git-stats hook(s) ( by something like `/(### git-stats hook \(begin\) ###.*### git-stats hook \(end\) ###)/mg`).

Comment (williamboman) [inline scripts/init-git-post-commit]: And now when the old script has been live for a while perhaps an additional check for the old syntax.

Comment (IonicaBizau) [inline scripts/init-git-post-commit]: Just a `grep` is enough I guess. Or maybe even asking the user if (s)he wants to override the file... That would be simpler.

Comment (williamboman) [inline scripts/init-git-post-commit]: I really don't like the idea of overwriting existing files, even after asking for permission.

Less experienced users or installations done in haste/negligence could potentially overwrite important hooks in the user's template dir, without them realizing it.

**Comment by williamboman** [comment]
What about now?

**Comment by IonicaBizau** [comment]
Big thanks! :+1:

**Pull request merged**